### DISCUSSION TITLE
Save current rows from table component to another SQL table on button click

### DISCUSSION THREAD

**Discussion by dado-ca** [Budibase/budibase#14035] [Help]
Hello everyone,

I have an issue hope somebody can help.

I have a table component in budibase, which populates its content dynamicly by executing an saved query.  The table component is bound to a DataProvider, the dataprovider is itself  executes the saved sql query.

The SELECT query, has some calculated fields, which are shown in the mentioned table component.

Now , I need to be able to Add button, when a user clicks on a button, it shoud fetch rows from the table component, and insert those rows in a separate postgers sql table. 

Now, because the table component gets its rows by executing an sql query i am not able to select rows and pluck those columns into an array with handlebars.

Is there another method to get it done. Thanks.

**Comment by mikesealey** [comment]
Hi @dado-ca 

This sounds like a great use-case for Automations! In your app, create an automation called that takes an array
![image](https://github.com/Budibase/budibase/assets/39269380/55399bed-5366-469d-998c-15e04ce50090)

Set your button to trigger the automation with the Data Provider's rows
![image](https://github.com/Budibase/budibase/assets/39269380/a32dde23-06db-4005-aa23-c92add9d4e8e)

You can then loop over the array passed into the automation, and use the individual elements in it to run a query.
![image](https://github.com/Budibase/budibase/assets/39269380/c0962d2b-63b4-4db2-9491-c5124df4e61a)

**Reply by dado-ca** [reply]
Thanks Mike, will give it a try. And post the results.

Hope this works, it would be great.
